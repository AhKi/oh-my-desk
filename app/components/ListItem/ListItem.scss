.ListItem {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 45px;
}

.active {
  opacity: 0.8;
}

.ListItem__button {
  width: 100%;
  height: 100%;
}