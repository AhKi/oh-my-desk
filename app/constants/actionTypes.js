export const STATUS_OPEN_PREFERENCE = 'STATUS_OPEN_PREFERENCE';
export const STATUS_CLOSE_PREFERENCE = 'STATUS_CLOSE_PREFERENCE';
export const STATUS_OPEN_WIDGET = 'STATUS_OPEN_WIDGET';
export const STATUS_CLOSE_WIDGET = 'STATUS_CLOSE_WIDGET';

export const SETTING_CHANGE_CURRENT_PAGE = 'SETTING_CHANGE_CURRENT_PAGE';
export const SETTING_CHANGE_FILTER = 'SETTING_CHANGE_FILTER';
export const SETTING_SELECT_WIDGET = 'SETTING_SELECT_WIDGET';

export const WIDGETS_REGISTER_NEW = 'WIDGETS_REGISTER_NEW';
export const WIDGETS_OPEN_WINDOW = 'WIDGETS_OPEN_WINDOW';
export const WIDGETS_CLOSE_WINDOW = 'WIDGETS_CLOSE_WINDOW';
export const WIDGETS_DELETE = 'WIDGETS_DELETE';

export const REGISTER_NEW_WIDGET = 'REGISTER_NEW_WIDGET';
