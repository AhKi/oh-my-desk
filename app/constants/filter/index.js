export const LATEST = 'LATEST';
export const OLDEST = 'OLDEST';
export const ACTIVATED = 'ACTIVATED';
