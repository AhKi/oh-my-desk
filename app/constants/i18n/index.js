import { remote } from 'electron';
import { langSelector } from 'store/share/status/selectors';
import contextMenu from './contextMenu';
import menu from './menu';
import preference from './preference';
import search from './search';
import update from './update';
import widget from './widget';

const getMatchedText = () => { // eslint-disable-line consistent-return
  let store;
  let lang;
  if (!remote) {
    const storeMain = require('store/storeMain').default; // eslint-disable-line global-require
    store = storeMain.getState();
    lang = langSelector(store);
  } else {
    store = JSON.parse(remote.getGlobal('getReduxState')());
    lang = store.status.lang; // eslint-disable-line prefer-destructuring
  }

  if (lang === 'English') {
    return {
      contextMenu: contextMenu.en,
      menu: menu.en,
      search: search.en,
      preference: preference.en,
      update: update.en,
      widget: widget.en,
    };
  }

  if (lang === 'Korean') {
    return {
      contextMenu: contextMenu.ko,
      menu: menu.ko,
      search: search.ko,
      preference: preference.ko,
      update: update.ko,
      widget: widget.ko,
    };
  }
};

export default getMatchedText;
