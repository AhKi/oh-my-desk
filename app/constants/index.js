export const NUMBER_PER_PAGE = 10; // eslint-disable-line
