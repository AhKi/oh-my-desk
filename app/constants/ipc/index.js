export const WIDGET_SHOW_INACTIVE = 'WIDGET_SHOW_INACTIVE';
export const WIDGET_INFO_REQUEST = 'WIDGET_INFO_REQUEST';
export const WIDGET_INFO_RESULT = 'WIDGET_INFO_RESULT';
export const WIDGET_MANAGE = 'WIDGET_MANAGE';
export const WIDGET_OPEN = 'WIDGET_OPEN';
