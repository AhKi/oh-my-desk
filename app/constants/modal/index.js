export const MAKE_WEB_WIDGET // eslint-disable-line
	= 'MAKE_WEB_WIDGET';