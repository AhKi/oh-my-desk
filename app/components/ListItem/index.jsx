import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import * as IPC from 'constants/ipc';
import './ListItem.scss';

const propTypes = {
  item: PropTypes.object, // eslint-disable-line
  selectedId: PropTypes.string,
  onSelectItem: PropTypes.func,
};

const defaultProps = {
  item: {},
  selectedId: '',
  onSelectItem() {},
};

class ListItem extends React.Component {
  constructor(props) {
    super(props);
    this.handleSelectList = this.handleSelectList.bind(this);
    this.handleWidgetOpen = this.handleWidgetOpen.bind(this);
  }

  handleSelectList() {
    this.props.onSelectItem(this.props.item.id);
  }

  handleWidgetOpen() {
    window.ipcRenderer.send(IPC.WIDGET_OPEN,
      Object.assign({}, this.props.item, { isActive: true }));
  }

  render() {
    const { item, selectedId } = this.props;
    const itemClassName = cx('ListItem', 'ListItem__button', {
      'ListItem__button-active': item.id === selectedId,
    });

    return (
      <li // eslint-disable-line
        className={itemClassName}
        onClick={this.handleSelectList}
        onDoubleClick={this.handleWidgetOpen}
      >
        {item.name}
      </li>
    );
  }
}

ListItem.propTypes = propTypes;
ListItem.defaultProps = defaultProps;

export default ListItem;
