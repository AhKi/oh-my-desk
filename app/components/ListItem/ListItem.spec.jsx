import React from 'react';
import { shallow } from 'enzyme';

import * as IPC from 'constants/ipc';
import ListItem from './';

describe('<ListItem />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<ListItem />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call correct when call handleSelectList', () => {
    const ipcRenderer = {
      send: () => {},
    };
    window.ipcRenderer = ipcRenderer;
    const send = jest.spyOn(window.ipcRenderer, 'send');
    const onSelectItem = jest.fn();
    const wrapper = shallow(
      <ListItem
        item={{
          id: 'mock-id',
        }}
        onSelectItem={onSelectItem}
      />,
    );

    wrapper.instance().handleSelectList();

    expect(onSelectItem).toHaveBeenCalledTimes(1);
    expect(onSelectItem).toHaveBeenCalledWith('mock-id');
    expect(send).toHaveBeenCalledTimes(1);
    expect(send).toHaveBeenCalledWith(
      IPC.WIDGET_SHOW_INACTIVE,
      'mock-id',
    );
  });

  it('should call correct when call handleWidgetOpen', () => {
    const ipcRenderer = {
      send: () => {
      },
    };
    window.ipcRenderer = ipcRenderer;
    const send = jest.spyOn(window.ipcRenderer, 'send');
    const wrapper = shallow(
      <ListItem
        item={{
          id: 'mock-id',
        }}
      />,
    );

    wrapper.instance().handleWidgetOpen();

    expect(send).toHaveBeenCalledTimes(1);
    expect(send).toHaveBeenCalledWith(
      IPC.WIDGET_OPEN,
      {
        id: 'mock-id',
        isActive: true,
      },
    );
  });
});
