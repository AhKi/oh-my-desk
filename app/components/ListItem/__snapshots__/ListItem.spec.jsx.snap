// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should match to snapshot when render default 1`] = `
<li
  className="ListItem ListItem__button"
  onClick={[Function]}
  onDoubleClick={[Function]}
/>
`;
